src/scales/scale.timeseries.js

```diff
@@ -110,6 +110,25 @@ class TimeSeriesScale extends TimeScale {
     return table;
   }
 
+  /**
+    * Generates all timestamps defined in the data.
+    * Important: this method can return ticks outside the min and max range, it's the
+    * responsibility of the calling code to clamp values if needed.
+    * @protected
+    */
+  _generate() {
+    const min = this.min;
+    const max = this.max;
+    let timestamps = super.getDataTimestamps();
+    if (!timestamps.includes(min) || !timestamps.length) {
+      timestamps.splice(0, 0, min);
+    }
+    if (!timestamps.includes(max) || timestamps.length === 1) {
+      timestamps.push(max);
+    }
+    return timestamps.sort((a, b) => a - b).map(x => +x);
```

Comment:
Should the cast to a number happen before the sort?

Reply:
First of all, let me say that I copied the same "method" used in time scale, where sort+map is the same.
IMO I'm not sure that the casting is correct on both implementation becauwe, if I'm not worng, it seems we are assuming that the date time is greater than Jan 1st 1970 (epoch).
Probably this casting should be removed and manage also negative epochs.

Reply:
Negative numbers are also numbers :)
![image](https://github.com/chartjs/Chart.js/assets/27971921/32d46edc-61da-4e2c-b54a-5a05fbd968c7)

Reply:
@kurkle I agree !!! but it's not clear to me why we are casting them to a positive number. This casting is also used in time scale... Maybe should be removed in time scale as well but I don't know hwy there is, maybe there is a reason. :(

Reply:
Its not casting to positive, just to number. Could use Number() instead or ~~ (a bit different though 😂) but + is shorter. I guess Number() would be transpiled to +.

Reply:
opss.... apologize, my knowledge on JS sometimes is not so good...anyway I don't know why it's needed, timestamps (epoch) should be always a number.

Reply:
Now I have understood the comment of @etimberg as well.
Tomorrow I will do in this PR changing also time scale or do you prefer a dedicated PR?

Reply:
Lets change it all together so we only have to test once

Reply:
@etimberg @kurkle @LeeLenaleee let me share here my analysis but the casting.

The time /timeseries scales are leveraging on `getDataTimestamps` or `getLabelTimestamps` functions to get the timestamps to use on the scale.

`getDataTimestamps` function is using `getAllParsedValues` method of the controller which is testing if the values in the data are "finite", therefore numbers.

`getLabelTimestamps` function is using `parse` function of the scale which is testing if the values in the data are "finite", therefore numbers.

Therefore I think that the casting should be removed.

Reply:
Guys, I need more time than expected because I'm checking also the functionalities of the options (like count, stepSize, maxTicksLimit and so on) comoaring with v2. Be patient...

Reply:
no worries @stockiNail, take as long as you need :)